libraries/authoritative_helpers.rb

```diff
@@ -17,41 +17,74 @@
+        },
+    ) unless defined? PdnsAuthoritativeResource::Helper::BACKENDS_PKG
 
-def backend_package_per_platform
-  return 'pdns-backend-geo'      if node['platform_family'] == 'debian' && new_resource.instance_name == 'geo'
-  return 'pdns-backend-ldap'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'ldap'
-  return 'pdns-backend-mysql '   if node['platform_family'] == 'debian' && new_resource.instance_name == 'mysql'
-  return 'pdns-backend-pgsql'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'postgresql'
-  return 'pdns-backend-pipe'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'pipe'
-  return 'pdns-backend-sqlite3'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'sqlite'
-  return 'pdns-backend-geoip'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'geoip'
-  return 'pdns-backend-lua'      if node['platform_family'] == 'debian' && new_resource.instance_name == 'lua'
-  return 'pdns-backend-mydns'    if node['platform_family'] == 'debian' && new_resource.instance_name == 'mydns'
-  return 'pdns-backend-odbc'     if node['platform_family'] == 'debian' && new_resource.instance_name == 'odbc'
-  return 'pdns-backend-opendbx'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'opendbx'
-  return 'pdns-backend-remote'   if node['platform_family'] == 'debian' && new_resource.instance_name == 'remote'
-  return 'pdns-backend-tinydns'  if node['platform_family'] == 'debian' && new_resource.instance_name == 'tinydns'
+    def backend_package_per_platform(name = nil)
+      v = BACKENDS_PKG[name]
+      case v
+      when Hash
+        v[node['platform_family']]
+      when NilClass
+        nil
+      else
+        v
+      end
+    end
 
-  return 'pdns-backend-geo'        if node['platform_family'] == 'rhel' && new_resource.instance_name == 'geo'
-  return 'pdns-backend-ldap'       if node['platform_family'] == 'rhel' && new_resource.instance_name == 'ldap'
-  return 'pdns-backend-lua'        if node['platform_family'] == 'rhel' && new_resource.instance_name == 'lua'
-  return 'pdns-backend-mydns'      if node['platform_family'] == 'rhel' && new_resource.instance_name == 'mydns'
-  return 'pdns-backend-mysql'      if node['platform_family'] == 'rhel' && new_resource.instance_name == 'mysql'
-  return 'pdns-backend-pipe'       if node['platform_family'] == 'rhel' && new_resource.instance_name == 'pipe'
-  return 'pdns-backend-postgresql' if node['platform_family'] == 'rhel' && new_resource.instance_name == 'postgresql'
-  return 'pdns-backend-remote'     if node['platform_family'] == 'rhel' && new_resource.instance_name == 'remote'
-  return 'pdns-backend-sqlite'     if node['platform_family'] == 'rhel' && new_resource.instance_name == 'sqlite'
-end
\ No newline at end of file
+    def default_authoritative_run_user
+      'pdns'
+    end
+
+    module_function
```

Comment:
This is required otherwise I would have to include `::Chef::Recipe.send(:include, PdnsAuthoritativeResource::Helpers)` in order to use `default_authoritative_config_directory` inside `test/cookbooks/pdns_test/recipes/authoritative_install_multi.rb`